integration/src/ethereum/ethereum.ts

```diff
@@ -119,6 +119,7 @@ export function ethereumTests(get: () => { wallet: core.HDWallet; info: core.HDW
       "ethSignTx() - ETH",
       async () => {
         if (!wallet) return;
+        if (!wallet.supportsOfflineSigning()) return;
```

Comment:
Metamask does not support offline signing, so this statement should force the test to return prematurely. There's a check used in a few places in the lib repository to take conditional action for wallets like metamask that do not support offline signing. Some or part of this check might be useful here instead.

```
  if (wallet.supportsOfflineSigning()) {
    try {
      // Call signTransaction and get signed transaction
    } catch (error) {
      throw new Error(error)
    }

    if (!signedTx) {
      throw new Error(error)
    }

    try {
      // broadcasat previously signed transaction
    } catch (error) {
      throw new Error(error)
    }

    return { txid }
  } else if (wallet.supportsBroadcast() && adapter.signAndBroadcastTransaction) {
    try {
      // Call signAndBroadcastTransaction to sign and broadcast in one step
    } catch (error) {
      throw new Error(error)
    }
    ```

Reply:
@pastaghost thanks for the detailed comment! I've updated the tests. 
Could you take another look, please?